### PR TITLE
Add parfor support for ndarray.fill.

### LINKED ISSUE #2742: Support ndarray.fill under parfors
`ndarray.fill` has parallel semantics and should be supported in ParallelAccelerator

### PR DESCRIPTION
**Pull request by DrTodd13**
Resolves #2742

### CODE REVIEW

**Comment by stuartarchibald** [comment]
@ehsantn any chance you could take a look at this please? Thanks.

**Comment by DrTodd13** [comment]
Used find_callname instead of data structure to remember getattrs.

**Comment by DrTodd13** [comment]
@ehsantn Thanks for the tip.  I also agree this is better.  I switched over.  Please have another look.

File: numba/parfor.py [hunk 1]
```diff
@@ -410,6 +410,30 @@ def linspace_3(start, stop, num):
     ('linspace', 'numpy'): linspace_parallel_impl,
 }
 
+def fill_parallel_impl(return_type, arr, val):
```
Comment (ehsantn) [inline numba/parfor.py]: `return_type` is not used.

Comment (DrTodd13) [reply numba/parfor.py]: It might be used by some other getattr function so I think we need to keep it as part of the API.

**Comment by stuartarchibald** [comment]
This needs conflicts resolving against master, thanks.

File: numba/parfor.py [hunk 2]
```diff
@@ -427,6 +427,30 @@ def linspace_3(start, stop, num):
...
+    if arr.ndim == 1:
+        def fill_1(in_arr, val):
+            numba.parfor.init_prange()
+            for i in numba.parfor.internal_prange(len(in_arr)):
+                in_arr[i] = val
+            return None
+    else:
+        def fill_1(in_arr, val):
+            numba.parfor.init_prange()
+            val = zero
```
Comment (stuartarchibald) [inline numba/parfor.py]: why zero?

Comment (stuartarchibald) [reply numba/parfor.py]: Is this branch tested?

Comment (DrTodd13) [reply numba/parfor.py]: Fixed and 2d test added.

**Comment by seibert** [comment]
Looks like the comments have been addressed and this is ready to merge.

**Pull request merged**